### PR TITLE
added quick note command input box

### LINKED ISSUE #4: Feature: Command add single note
This command will open a input dialogue to input a single note that will be added to the daily note file.

### PR DESCRIPTION
**Pull request by javalight**
Input box working, adds new note to line 2 of file, can be improved.

bug: if date is old and new note is added, new note will not show up, async issue.

The fs library is old and has been updated to fs/promises with async await, because of the fs librarys callback style it would be ugly to fix this bug, need to just change everything over to fs/promises and life will be easy. 

fix: fs is old library, move everything to fs promises/async await
https://dev.to/mrm8488/from-callbacks-to-fspromises-to-handle-the-file-system-in-nodejs-56p2

### CODE REVIEW

**Comment by javalight** [comment]
i can make a commit to this pull-request changing everything to async await, waiting for permission :)

**Comment by inaki-ibarra** [comment]
> i can make a commit to this pull-request changing everything to async await, waiting for permission :)

Sorry for the late reply, I got busy. Yeah, you can change the FS to `async-await`, it seems the right way moving forward.

File: extension.js [hunk 1]
```diff
@@ -107,8 +135,24 @@ function activate(context) {
     });
   }
 
+  function appendToFileAtLine(filePath, content, lineNumber, callback) {
+    fs.readFile(filePath, "utf8", function (error, result) {
+      if (error && error.code !== "ENOENT") {
```
Comment (inaki-ibarra) [inline extension.js:131]: I think you forgot to trigger the error handler here (calling the callback).

File: package.json [hunk 2]
```diff
@@ -18,14 +18,19 @@
...
   "main": "./extension.js",
   "contributes": {
     "commands": [
       {
         "command": "extension.open",
         "title": "Open Daily Notes"
+      },
+      {
+        "command": "extension.addQuickNote",
+        "title": "Add Quick Single Note"
```
Comment (inaki-ibarra) [inline package.json]: I really like to use "Daily Notes" here to make things consistent but I'm not sure how. Any thoughts?

File: package.json [hunk 3]
```diff
@@ -18,14 +18,19 @@
     "Other"
   ],
   "activationEvents": [
-    "onCommand:extension.open"
+    "onCommand:extension.open",
+    "onCommand:extension.addQuickNote"
```
Comment (inaki-ibarra) [inline package.json]: "Add" seems a good keyword, What do you think of "Append" (like in IOS notes) or "Insert".

**Comment by inaki-ibarra** [comment]
Thanks for submitting this feature. Aside from my comments on this PR, I also suggest adding a key-bind for appending notes. Maybe something similar to Alt+N.

File: package.json (hunk 3 above)
Comment (javalight) [reply package.json]: Love it all, went with insert for now tell me how it looks

File: package.json (hunk 2 above)
Comment (javalight) [reply package.json]: changes names to dailyNotes.insert

**Comment by javalight** [comment]
Bug fixed, everything working.

**Comment by inaki-ibarra** [comment]
> Bug fixed, everything working.

Thanks for this, I'll review and test in my local this weekend. Hopefully, I can release a new version.

**Comment by inaki-ibarra** [comment]
LGTM, I'll upload the new version after I update the README. Thanks!

**Pull request merged**